src/rust/lib/graph-editor/src/component/visualization.rs

```diff
@@ -0,0 +1,332 @@
+
+use ensogl::display::DomSymbol;
+use ensogl::display::object::class::Object;
+use ensogl::display::object::class::ObjectOps;
+use ensogl::display;
+use ensogl::system::web;
+use web::StyleSetter;
+
+
+// ============================================
+// === Wrapper for Visualisation Input Data ===
+// ============================================
+
+/// Wrapper for data that can be consumed by a visualisation.
+/// TODO replace with better typed data wrapper.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub enum Data {
+    JSON { content : String },
+    Empty,
+}
+
+impl Data {
+    /// Render the data as JSON.
+    pub fn as_json(&self) -> String {
+        match &self {
+            Data::JSON { content } => content.clone(),
+            Data::Empty => { "{}".to_string() },
+        }
+    }
+}
+
+impl Default for Data{
+    fn default() -> Self {
+        Data::Empty
+    }
+}
+
+
+
+// =============================================
+// === Internal Visualisation Representation ===
+// =============================================
+
+/// Inner representation of a visualisation.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Visualization {
+    content : Rc<DomSymbol>
+}
+
+impl Object for Visualization {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.content.display_object()
+    }
+}
+
+impl Visualization {
+    /// Update the visualisation with the given data.
+    // TODO remove dummy functionality and use an actual visualisation
+    pub fn set_data(&self, data:Data){
+                self.content.dom().set_inner_html(
+                    &format!(r#"
+<svg>
+  <circle style="fill: #69b3a2" stroke="black" cx=50 cy=50 r={}></circle>
+</svg>
+"#, data.as_json()));
+     }
+
+    /// Make the visualisation visible in the scene.
+    pub fn show(&self) {
+        // TODO investigate how to do this through display_object functionality.
+        self.content.dom().set_style_or_panic("visibility", "visible");
+    }
+
+    /// Hide the visualisation in the scene.
+    pub fn hide(&self) {
+        // TODO investigate how to do this through display_object functionality.
+        self.content.dom().set_style_or_panic("visibility", "hidden");
+    }
+}
+
+impl From<DomSymbol> for Visualization {
+    fn from(symbol: DomSymbol) -> Self {
+        Visualization { content : Rc::new(symbol) }
+    }
+}
+
+impl From<Rc<DomSymbol>> for Visualization {
+    fn from(symbol: Rc<DomSymbol>) -> Self {
+        Visualization { content : symbol }
+    }
+}
+
+
+
+// ============================
+// === Visualization Events ===
+// ============================
+
+/// Visualization events.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Events {
+    pub network           : frp::Network,
+    pub show              : frp::Source,
+    pub hide              : frp::Source,
+    pub toggle_visibility : frp::Source,
+    pub update_visualization    : frp::Source<Option<Visualization>>,
```

Comment:
Im not sure if "update" is the right word here. It took me a while to understand that this one replaces the visualization. You know "update visualization" might mean to change the data it displays. I would name it `replace_visualization` instead, but take it as a personal preference and if you have a strong pref to leave it as it, go ahead and ignore this comment :)